erts/emulator/nifs/common/prim_file_nif.c

```diff
@@ -521,6 +524,61 @@ static ERL_NIF_TERM open_nif(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]
     return enif_make_tuple2(env, am_ok, result);
 }
 
+static ERL_NIF_TERM file_desc_to_ref_nif(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
+    posix_errno_t posix_errno;
+    efile_data_t *d;
+
+    ErlNifPid controlling_process;
+    enum efile_modes_t modes;
+    ERL_NIF_TERM result;
+    int fd;
+
+    ASSERT(argc == 2);
+    if(!enif_is_list(env, argv[1])) {
+        return enif_make_badarg(env);
+    }
+
+    if(!enif_is_number(env, argv[0])) {
+        return enif_make_badarg(env);
+    }
+
+    modes = efile_translate_modelist(env, argv[1]);
```

Comment:
The `fd` is already open, why require a mode list?

Reply:
I thought it could be useful to check that the FD has the expected access flags but I have now realized that one does not get much benefit from that (at least not in how it is used now) as one will notice that anyway when one starts to use the file descriptor. So I think I will remove the mode list parameter.